### ISSUE TITLE
[BUG] To property of ServiceBusMessage in Azure.Messaging.ServiceBus v7.1.2 cannot be null

### ISSUE THREAD

**Issue by petrformanek** [Azure/azure-sdk-for-net#21689]
**Describe the bug**
Dear Azure SDK team,
`ServiceBusMessage.To` property cannot have null value. It is a breaking change against `Microsoft.Azure.ServiceBus` v5.1.2. This breaking change complicates our effort to migrate to the new SDK. We use `To` property in our SQL rules like this: `$"AND (sys.To IS NULL OR sys.To = '{subscriptionSettings.SubscriptionName}')";`.
Now I never get `NULL` in the `To` property when the message is sent by the new SDK so I will never get messages to my subscription.

**Expected behavior**
```csharp
var message = new ServiceBusMessage();
Assert.IsNull(message.To);
```

**Actual behavior (include Exception or Stack Trace)**
```csharp
var message = new ServiceBusMessage();
Assert.AreEqual(message.To, "");
```
Underlying object has nullable struct `AmqpAddress` for `To` property but `Nullable<T>.ToString()` returns an empty string when null.

**To Reproduce**
As described above.

**Environment:**
 - Name and version of the Library package used: `Azure.Messaging.ServiceBus v7.1.2`

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by JoshLove-msft** [comment]
Hi @petrformanek,
We're sorry you are running into this issue. As the library is already GA, changing the To property to return null in this situation could break some users. Is it possible for you to update your SQL filter to check for empty string rather than null?

**Comment by petrformanek** [comment]
Hi @JoshLove-msft ,
yes I already change the filter to check for empty string. But we have many consumers across various teams. So this inability to set null makes new SDK adoption much complicated.
What about just being able to set null explicitly e.g. `message.To = null`?

**Comment by JoshLove-msft** [comment]
Do you mean that other teams would also have to do a similar filter update and that would be difficult to do?

**Comment by JoshLove-msft** [comment]
> What about just being able to set null explicitly e.g. message.To = null?

I think this may be possible for us to support. I will discuss with the team and see what they think.

**Comment by petrformanek** [comment]
> Do you mean that other teams would also have to do a similar filter update and that would be difficult to do?

We have our messaging SDK which wraps Service Bus SDK. The filter is a part of our SDK which is used by our teams to consume messages from service bus. 
We have just two message producers which we cannot upgrade to the latest SDK until all consumers upgrade to our latest SDK (with fixed filters).
It just takes some weeks or months to force all the teams to upgrade.

**Comment by JoshLove-msft** [comment]
I talked this over with the team and since the library is already GA, unfortunately we can't change the behavior to return null when the property is not set. I think the best path forward would be to update the consumers to filter based on an empty string.

**Comment by petrformanek** [comment]
Ok, thanks for your information.

**Comment by petrformanek** [comment]
Hi @JoshLove-msft,
It looks like that you decide to fix it in [7.5](https://github.com/Azure/azure-sdk-for-net/blob/Azure.Messaging.ServiceBus_7.5.0/sdk/servicebus/Azure.Messaging.ServiceBus/CHANGELOG.md#breaking-changes). Can you please confirm?

**Comment by JoshLove-msft** [comment]
> Hi @JoshLove-msft, It looks like that you decide to fix it in [7.5](https://github.com/Azure/azure-sdk-for-net/blob/Azure.Messaging.ServiceBus_7.5.0/sdk/servicebus/Azure.Messaging.ServiceBus/CHANGELOG.md#breaking-changes). Can you please confirm?

Yep, sorry I missed updating this issue - this has been fixed.
